test/CMakeLists.txt

```diff
@@ -446,12 +461,15 @@ ngen_add_test(
         NGen::core_mediator
         NGen::forcing
         NGen::geojson
+        NGen::geopackage
         NGen::realizations_catchment
         NGen::mdarray
         NGen::mdframe
         NGen::logging
         NGen::ngen_bmi
         testbmicppmodel
+    REQUIRES
+        NGEN_WITH_SQLITE
```

Comment:
I'm concerned about making the entire `test_unit` build dependent on having SQLite support enabled.

Reply:
Maybe the test case could say

```
         #if !defined(NGEN_WITH_SQLITE3)
            GTEST_SKIP() << "Compiled without SQLite support"
         #endif
```
?

Reply:
The problem is with the ```geopackage``` is involved in the test code.

Reply:
Haven't reviewed this, but IIRC from an earlier discussion, you could take out the GeoPackage reading, and just use GeoJSON, since it isn't needed to test the functionality of this PR.